lib/util.py

```diff
@@ -327,7 +327,24 @@ def timeInDayLocalSeconds():
     return int(time.time() - sod)
 
 
-time_format_twentyfour = "M" not in xbmc.getInfoLabel('System.Time')
+def get24hFormat():
+    """
+    This takes the 24h setting from Kodi and tries to determine whether the user wants the 24h or 12h time format.
+    :return:
+    """
+    try:
+        use_24h = rpc.Settings.GetSettingValue(setting="locale.use24hourclock")["value"]
+    except:
```

Comment:
If this will often fail (ex. because there isn't always a `"value"` key), let's put a specific except clause for the error (ex. `KeyError`) and then maybe still have a blanket `except:` but call `Error()` there so it is at least logged. If the exception is just there for safety, then we can just add the `Error()` call.

Reply:
It's there for safety, yes, in case the RPC doesn't answer in time or the result doesn't have the `value` key. I can add a separate check for that case and log both errors.

Reply:
Basically if we're adding a check that we know is going to fail often for a known reason, we don't want to log an error for that, we can (if needed) just `DEBUG_LOG('message')`. But if we are adding a fail-safe to catch an unexpected error then we should use `ERROR()` which will log whether debug logging is on (so we don't want to use it for known failures, because it scares users :smile:). I (for example) often use ERROR() for network requests that I expect to work 99% of the time, but as they are network requests have the potential to fail, and I want the traceback in those cases even if I don't plan on adding any extra handling.

I think in this case just adding ERROR() to the except should be fine, because I think the expectation is that `value` should always be there and the rpc call should always return without issue.